models/reports.go

```diff
@@ -1,11 +1,19 @@
 package models
 
-func (rl ReportListingChild) GetKind() string { return rl.Kind }
+func (rl ReportListingChild) GetKind() string               { return rl.Kind }
+func (rl *ReportListing) GetChildren() []ReportListingChild { return rl.Data.Children }
 func (rl ReportListingChild) GetId() string {
-	if rl.GetKind() == "t1" {
-		return rl.Data.(CommentListingDataChildrenData).Name
-	} else {
-		return rl.Data.(PostListingChildData).Name
+	return rl.Data.Name
+}
+func (rl *ReportListingChild) GetUserReports() []UserReport {
+	reports := make([]UserReport, 0)
+	for i := range rl.Data.UserReports {
+		report := UserReport{}
+		report.Reason = rl.Data.UserReports[i].([]interface{})[0].(string)
+		report.NumOfReports = rl.Data.UserReports[i].([]interface{})[1].(float64)
+		report.SnoozeStatus = rl.Data.UserReports[i].([]interface{})[2].(bool)
+		report.CanSnooze = rl.Data.UserReports[i].([]interface{})[3].(bool)
+		reports = append(reports, report)
```

Comment:
I like this casting, great job! It also simplified some getters and got rid of the if statement there as we verify the type before then